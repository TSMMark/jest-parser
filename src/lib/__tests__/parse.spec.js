import { readFileSync } from 'fs'
import path from 'path'
import {
    parse,
    getStringIndent,
    unIndentString,
    getFnBlockedContent
} from '../parse'

const jestFileFixture = readFileSync(path.join(__dirname, './fixtures/jest.test.js')).toString()

describe('getStringIndent()', () => {
    it('returns the minimum amount of indentation found in given text', () => {
        const indent = getStringIndent(`  abc\n    123\n  do re mi`)
        expect(indent).toEqual(2)
    })
})

describe('unIndentString()', () => {
    it('removes the minimum amount of indentation found in given text', () => {
        const unIndented = unIndentString(`  abc\n    123\n  do re mi`)
        expect(unIndented).toEqual(`abc\n  123\ndo re mi`)
    })
})

describe('getFnBlockedContent()', () => {
    it('retrieves metadata about a particular set of describe functions', () => {
        const parsedTree = getFnBlockedContent(jestFileFixture, ['describe', 'it'])
        expect(parsedTree).toMatchSnapshot()
    })
})

describe('parse()', () => {
    it('retrieves metadata about a jest source', () => {
        const parsedTree = parse('Some jest file', jestFileFixture)
        expect(parsedTree).toMatchSnapshot()
    })
})
