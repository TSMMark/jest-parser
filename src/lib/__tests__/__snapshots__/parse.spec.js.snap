// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFnBlockedContent() retrieves metadata about a particular set of describe functions 1`] = `
Object {
  "describe": Array [
    Object {
      "code": "// some code 1
describe('Nested describe 1', () => {
    it('does this and that', () => {
        // code of what it does
        expect(true).toBeTruthy()
    })
})",
      "describe": Array [
        Object {
          "code": "it('does this and that', () => {
    // code of what it does
    expect(true).toBeTruthy()
})",
          "describe": Array [],
          "end": 6,
          "it": Array [
            Object {
              "code": "// code of what it does
expect(true).toBeTruthy()",
              "describe": Array [],
              "end": 3,
              "it": Array [],
              "start": 0,
              "title": "does this and that",
            },
          ],
          "start": 1,
          "title": "Nested describe 1",
        },
      ],
      "end": 8,
      "it": Array [],
      "start": 0,
      "title": "Some describe title 1",
    },
    Object {
      "code": "// some code 2",
      "describe": Array [],
      "end": 12,
      "it": Array [],
      "start": 10,
      "title": "Some describe title 2",
    },
    Object {
      "code": "// some code 3",
      "describe": Array [],
      "end": 16,
      "it": Array [],
      "start": 14,
      "title": "Some describe title 3",
    },
  ],
  "it": Array [],
}
`;

exports[`parse() retrieves metadata about a jest source 1`] = `
Object {
  "describe": Array [
    Object {
      "code": "// some code 1
describe('Nested describe 1', () => {
    it('does this and that', () => {
        // code of what it does
        expect(true).toBeTruthy()
    })
})",
      "describe": Array [
        Object {
          "code": "it('does this and that', () => {
    // code of what it does
    expect(true).toBeTruthy()
})",
          "describe": Array [],
          "end": 6,
          "it": Array [
            Object {
              "code": "// code of what it does
expect(true).toBeTruthy()",
              "describe": Array [],
              "end": 3,
              "it": Array [],
              "start": 0,
              "test": Array [],
              "title": "does this and that",
            },
          ],
          "start": 1,
          "test": Array [],
          "title": "Nested describe 1",
        },
      ],
      "end": 8,
      "it": Array [],
      "start": 0,
      "test": Array [],
      "title": "Some describe title 1",
    },
    Object {
      "code": "// some code 2",
      "describe": Array [],
      "end": 12,
      "it": Array [],
      "start": 10,
      "test": Array [],
      "title": "Some describe title 2",
    },
    Object {
      "code": "// some code 3",
      "describe": Array [],
      "end": 16,
      "it": Array [],
      "start": 14,
      "test": Array [],
      "title": "Some describe title 3",
    },
  ],
  "it": Array [],
  "test": Array [],
  "title": "Some jest file",
}
`;
